src/main/java/seedu/taassist/model/student/MappedStudentViewList.java

```diff
@@ -0,0 +1,209 @@
+package seedu.taassist.model.student;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
+import javafx.collections.transformation.TransformationList;
+import seedu.taassist.model.moduleclass.ModuleClass;
+import seedu.taassist.model.session.Session;
+
+/*
+  Adapted from James_D's answer on Stack Overflow, see
+  https://stackoverflow.com/questions/32294055/mirror-one-observablelist-to-another.
```

Comment:
Can change with the direct link: 
```suggestion
  https://stackoverflow.com/a/32297235.
```